### PR TITLE
Optimized query to update cards from proposals

### PR DESCRIPTION
**Pull request by Devorein**

### CODE REVIEW

File: lib/focalboard/updateCardsFromProposals.ts [hunk 1]
```diff
@@ -27,6 +33,93 @@ import { generateResyncedProposalEvaluationForCard } from './generateResyncedPro
...
+  cursor?: string | null;
+  spaceId: string;
+}) {
+  const batchSize = 100;
+  const proposalPages = await prisma.page.findMany({
+    where: {
+      spaceId,
+      type: 'proposal',
+      proposal: {
+        status: {
```
Comment (mattcasey) [inline lib/focalboard/updateCardsFromProposals.ts:99]: might be more optimal to say status: "published"

Comment (Devorein) [reply lib/focalboard/updateCardsFromProposals.ts:99]: fair point, updated.

File: lib/focalboard/updateCardsFromProposals.ts [hunk 2]
```diff
@@ -129,33 +125,35 @@ export async function updateCardsFromProposals({
...
-    select: {
-      permissions: true
-    }
-  });
-
-  const proposalBoardBlock = (await prisma.proposalBlock.findUnique({
-    where: {
-      id_spaceId: {
-        id: DEFAULT_BOARD_BLOCK_ID,
+  const [board, rootPagePermissions, proposalBoardBlock] = await prisma.$transaction([
```
Comment (mattcasey) [inline lib/focalboard/updateCardsFromProposals.ts:128]: do these in parallel

File: lib/focalboard/updateCardsFromProposals.ts [hunk 3]
```diff
@@ -69,12 +69,7 @@ const pageSelectObject = {
           id: true
         }
       },
-      fields: true,
-      form: {
-        select: {
-          id: true
-        }
-      }
+      fields: true
```
Comment (mattcasey) [inline lib/focalboard/updateCardsFromProposals.ts:72]: we didnt need to "JOIN" the form table, we alerady get the formId column

Comment (Devorein) [reply lib/focalboard/updateCardsFromProposals.ts:72]: good catch

**Pull request merged**